samples/SortBenchmark/DiskHelper.fs

```diff
@@ -291,3 +299,85 @@ type DiskHelper(records:int64) =
                 x.sorteddataSP.Save() 
                 b <- true
                 x.sorteddataSP.lock := 0
+
+
+
+[<AllowNullLiteral>]
+type RollingFileMgr(records:int64) = 
+    let prefix = "sortFile_"
+    let dirs = [|@"C:\sortbenchmark";@"D:\sortbenchmark\";@"E:\sortbenchmark\";@"F:\sortbenchmark\"|]
+    member val dirIndex = ref -1 with get,set
+
+    member val writeFileLock = Array.create dirs.Length (ref 0)
+    member val MaxFileSize = 10000000
+    
+    member val fileHandleQ = new ConcurrentQueue<(FileStream*int)>()
+
+    member val readyFileQ = new ConcurrentQueue<string>()
+    member val fileNameQ = new ConcurrentQueue<string>()
+
+
+    member x.Init() =
+        for i = 0 to dirs.Length - 1 do
+            let dir = Path.Combine(dirs.[i],prefix+(string) records)
+            if not (Directory.Exists(dir)) then
+                Directory.CreateDirectory(dir) |> ignore
+
+            //x.fileHandleQ.Enqueue(x.CreateNewFile() , 0)
+            
+        ()
+
+    member x.GetNewFilename() =
+        let index = Interlocked.Increment(x.dirIndex)
+        let dir = Path.Combine(dirs.[index % dirs.Length],prefix+(string) records)
+        Path.Combine(dir, (string) index + ".bin")
+
+    member x.CreateNewFile() =
+        let fp = x.GetNewFilename()
+        x.fileNameQ.Enqueue(fp)
+        new FileStream(fp,FileMode.Create)
+
+
+    member x.EnqueueFileHandle(fh:FileStream, size:int) =
+        x.fileHandleQ.Enqueue(fh,size)
+
+
+
+[<AllowNullLiteral>]
+type SingleThreadExec1() =
```

Comment:
There's almost never a reason for a type with an empty constructor to have an [<AllowNullLiteral>]. To delay initialization, we can always just use a Lazy<'T>, or just create the object if initialization is not an issue.